src/import/import-compress.c

```diff
@@ -231,6 +290,55 @@ int import_compress_init(ImportCompress *c, ImportCompressType t) {
                 break;
 #endif
 
+#if HAVE_ZSTD
+        case IMPORT_COMPRESS_ZSTD: {
+                size_t r2;
+                int ncpus;
+
+                assert(c->zstd_c == NULL);
+                c->zstd_c = ZSTD_createCStream();
+                if (c->zstd_c == NULL)
+                        return -ENOMEM;
+
+                /* TODO: better default? zstd -3 is really weak */
+                r2 = ZSTD_CCtx_setParameter(c->zstd_c, ZSTD_c_compressionLevel, ZSTD_CLEVEL_DEFAULT);
+                if (ZSTD_isError(r2))
+                        return log_error_errno(
+                                SYNTHETIC_ERRNO(EIO),
+                                "Failed to set zstd compression level: %s",
+                                ZSTD_getErrorName(r2)
+                        );
+
+                r2 = ZSTD_CCtx_setParameter(c->zstd_c, ZSTD_c_checksumFlag, 1);
+                if (ZSTD_isError(r2))
+                        return log_error_errno(
+                                SYNTHETIC_ERRNO(EIO),
+                                "Failed to enable zstd output checksumming: %s",
+                                ZSTD_getErrorName(r2)
+                        );
+
+                ncpus = cpus_in_affinity_mask();
+                if (ncpus > 0) {
+                        r2 = ZSTD_CCtx_setParameter(c->zstd_c, ZSTD_c_nbWorkers, ncpus);
+                        if (!ZSTD_isError(r2))
+                                log_debug("Enabled zstd multithreaded compression with %d threads", ncpus);
+                        else
+                                log_warning(
+                                        "Failed to enable zstd multithreaded compression with %d threads, ignoring: %s",
+                                        ncpus,
+                                        ZSTD_getErrorName(r2)
+                                );
+                } else if (ncpus < 0)
+                        log_warning_errno(ncpus,
+                                "Failed to determine available CPUs, not enabling zstd multithreaded compression: %m");
+                else
```

Comment:
the call will never return 0. you can catch this case with assert_not_reached() or some other assert()